mongo/collection.go

```diff
@@ -699,3 +699,18 @@ func (coll *Collection) Watch(ctx context.Context, pipeline interface{},
 func (coll *Collection) Indexes() IndexView {
 	return IndexView{coll: coll}
 }
+func (coll *Collection) Drop(ctx context.Context) error {
+	if ctx == nil {
+		ctx = context.Background()
+	}
+
+	cmd := command.DropCollection{
+		DB:         coll.db.name,
+		Collection: coll.name,
+	}
+	_, err := dispatch.DropCollection(ctx, cmd, coll.client.topology, coll.writeSelector)
+	if err != nil {
+		return err
```

Comment:
We just need to check if the error is a `command.Error` and if it is if the `Code` property is equal to `26`, if it is, return nil.